qa327/frontend.py

```diff
@@ -0,0 +1,186 @@
+from flask import render_template, request, session, redirect
+from qa327 import app
+import qa327.backend as bn
+import re
+
+"""
+This file defines the front-end part of the service.
+It elaborates how the services should handle different
+http requests from the client (browser) through templating.
+The html templates are stored in the 'templates' folder. 
+"""
+
+
+@app.route('/register', methods=['GET'])
+def register_get():
+    # templates are stored in the templates folder
+    return render_template('register.html', message='')
+
+
+@app.route('/register', methods=['POST'])
+def register_post():
+    email = request.form.get('email')
+    name = request.form.get('name')
+    password = request.form.get('password')
+    password2 = request.form.get('password2')
+    error_message = None
+
+
+    if password != password2:
+        error_message = "The passwords do not match"
+
+    elif not is_valid_email(email):
```

Comment:
I like this structure a lot better than having each individual condition assessed in this one function. Makes sense to me!